client/src/components/Tool/ToolForm.vue

```diff
@@ -2,7 +2,7 @@
     <ConfigProvider v-slot="{ config }">
         <CurrentUser v-slot="{ user }">
             <UserHistories v-if="user" :user="user" v-slot="{ currentHistory }">
```

Comment:
If all you need is the currentHistoryId you could just use that slot. It should be coming from vuex, so normally there shouldn't be a delay ... which I think isn't necessary for the id. Maybe try that in combination with:
```diff
diff --git a/client/src/components/History/providers/UserHistories/UserHistories.js b/client/src/components/History/providers/UserHistories/UserHistories.js
index 56b3f761eb..e9547e7db0 100644
--- a/client/src/components/History/providers/UserHistories/UserHistories.js
+++ b/client/src/components/History/providers/UserHistories/UserHistories.js
@@ -65,7 +65,7 @@ export default {
 
             // currently selected history object, should be a full object not just a summary
             currentHistory: this.currentHistoryModel,
-            currentHistoryId: this.currentHistoryModel?.id || null,
+            currentHistoryId: this.currentHistoryId,
 
             handlers: {
                 // Updates the history in the store without a trip to the server, in the event that a

```

Reply:
Thanks @mvdbeek.